evalml/pipelines/components/transformers/imputers/per_column_imputer.py

```diff
@@ -61,7 +66,9 @@ def fit(self, X, y=None):
         """
         X = infer_feature_types(X)
         self.imputers = dict()
-        for column in X.columns:
+
+        columns_to_impute = X.columns if self.impute_all else self.impute_strategies
```

Comment:
Does this work correctly even though `X.columns` is a list and `self.impute_strategies` is a dict? Why not use `self.impute_strategies.keys()`?

Reply:
Heh, I think here it works because we just use it to iterate, and iterating over a dictionary will use its keys! But happy to update this to be more clear / in case we update this logic :)